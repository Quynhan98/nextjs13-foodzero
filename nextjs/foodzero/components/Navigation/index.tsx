import { memo } from 'react'
import { Box, BoxProps, ListItem, UnorderedList } from '@chakra-ui/react'
import Link from 'next/link'

type NavigationType = {
  value: number
  name: string
  href: string
}

interface NavigationProps extends BoxProps {
  navList: NavigationType[]
  onClick?: () => void
}

const Navigation = ({ navList, onClick, ...rest }: NavigationProps) => {
  return (
    <Box as="nav" {...rest}>
      <UnorderedList
        display="flex"
        flexDirection="column"
        gap={{ base: '15px', '2xl': '28px' }}
      >
        {navList.map((obj) => (
          <ListItem
            key={obj.value}
            color="white"
            textTransform="uppercase"
            fontSize={{ base: '26px', '2xl': '66px' }}
            lineHeight={{ base: '30px', '2xl': '82px' }}
            fontWeight="bold"
          >
            <Link href={obj.href} onClick={onClick} prefetch={false}>
              {obj.name}
            </Link>
          </ListItem>
        ))}
      </UnorderedList>
    </Box>
  )
}

export default memo(Navigation)
