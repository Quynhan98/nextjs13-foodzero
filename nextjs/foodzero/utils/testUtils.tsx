import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { BookingProvider } from '@contexts/BookingProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <AuthProvider>
        <BookingProvider>{component}</BookingProvider>
      </AuthProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
