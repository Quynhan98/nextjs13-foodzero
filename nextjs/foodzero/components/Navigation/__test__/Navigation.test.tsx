import { render, screen } from '@utils/testUtils'

// Constants
import { NAV_LIST } from '@constants/index'

// Components
import Navigation from '@components/Navigation'

describe('Navigation component', () => {
  it('Renders Navigation correctly', () => {
    const { container } = render(<Navigation navList={NAV_LIST} />)

    expect(container).toMatchSnapshot()
  })

  it('Should render a sufficient number of items', () => {
    render(<Navigation navList={NAV_LIST} />)

    const items = screen.getAllByRole('listitem')

    expect(items.length).toBe(NAV_LIST.length)
  })
})
