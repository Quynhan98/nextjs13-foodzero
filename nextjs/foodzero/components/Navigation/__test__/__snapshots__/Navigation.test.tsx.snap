// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component Renders Navigation correctly 1`] = `
<div>
  <nav
    class="css-0"
  >
    <ul
      class="css-1hgnqmf"
      role="list"
    >
      <li
        class="css-18dlst6"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="css-18dlst6"
      >
        <a
          href="/menu"
        >
          Menu
        </a>
      </li>
      <li
        class="css-18dlst6"
      >
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li
        class="css-18dlst6"
      >
        <a
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
